src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -151,6 +156,40 @@
     return ResponseEntity.ok(buildDetailedCaseDTO(caseObj, caseevents, iacFlag));
   }
 
+  /**
+   * the POST endpoint to generate and return a new iac
+   *
+   * @param collectionexerciseid to generate iac for
+   * @param ruref to generate iac for
+   * @return the case created
+   * @throws CTPException something went wrong
+   */
+  @RequestMapping(value = "/iac/{collectionexerciseid}/{ruref}", method = RequestMethod.POST)
+  public ResponseEntity<CaseDetailsDTO> generateNewIac(@PathVariable("collectionexerciseid") final UUID collectionexerciseid,
+                                                       @PathVariable("ruref") final String ruref)
+          throws CTPException {
+    log.info("Entering generateNewIac with collectionexerciseid {} and ruref {}", collectionexerciseid, ruref);
+
+    CaseGroup caseGroup = caseGroupService.findCaseGroupByCollectionExerciseIdAndRuRef(collectionexerciseid, ruref);
+    if (caseGroup == null) {
+      throw new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND,
+              String.format("CaseGroup not found for collectionexerciseid %s and ruref %s",
+                      collectionexerciseid, ruref));
+    }
+    List<Case> casesList = caseService.findCasesByCaseGroupFK(caseGroup.getCaseGroupPK());
+    Case latestCase = casesList.get(casesList.size() - 1);
+    List<String> iacs = internetAccessCodeSvcClientService.generateIACs(1);
+
+    SampleUnitBase sampleUnitBase = new SampleUnitBase(caseGroup.getSampleUnitRef(), Character.toString('B'),
+            caseGroup.getPartyId().toString(), latestCase.getCollectionInstrumentId().toString());
+    Case newCase = caseService.generateNewCase(sampleUnitBase, caseGroup, iacs.get(0), latestCase.getActionPlanId());
+    if (newCase == null) {
```

Comment:
newCase will never be null. If there is an error when saving to the database it would come from a DataAccessException. https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/dao/DataAccessException.html